deepreg/registry.py

```diff
@@ -74,11 +74,50 @@ def decorator(_cls):
 
         return decorator
 
+    def build_from_config(self, category: str, config: dict, default_args=None):
```

Comment:
this is probably the place to add a few checks to produce back-compatible error messages? you can just check a few old ones, e.g. energy_type, loss_type, loss_scale... then i think it should be pretty safe to recognise the old format config files.

Reply:
I strongly object the back compatibility.

Reply:
The whole idea is to make it simple. I will do meaningful error messages.

Reply:
> The whole idea is to make it simple. I will do meaningful error messages.

i didn't mean back-compatible code....